src/atoms/Tab/Tab.native.js

```
@@ -0,0 +1,133 @@
+import automation from '../../_helpers/automation-attributes';
+
+const styles = {
+  label: {
+    borderBottomWidth({ isActive }) {
+      if (isActive) {
+        return '1px';
+      }
+      return '0px';
+    },
+    borderBottomColor({ isActive, theme, disabled }) {
+      if (disabled) {
+        if (isActive) {
+          return getColor(theme, 'primary.300');
+        }
+      }
+      if (isActive) {
+        return getColor(theme, 'primary.800');
+      }
+      return 'transparent';
+    },
+    margin() {
+      return [0, 0, 0, 0.5];
+    },
+  },
+};
+
+const StyledTabButton = styled(TouchableOpacity)(
+  (props) => `
+min-width: 48px;
+background-color: ${props.backgroundColor};
+border-top-left-radius: 2px;
+border-top-right-radius: 2px;
+`,
+);
+
+const Label = styled(View)`
+  padding: 8px 0px;
+  border-bottom-width: ${styles.label.borderBottomWidth};
+  border-bottom-color: ${styles.label.borderBottomColor};
+`;
+
+const getLabelColor = ({ isActive, disabled }) => {
+  let color = 'shade.960';
+
+  if (isActive) {
+    color = disabled ? 'primary.700' : 'primary.800';
+  } else {
+    color = disabled ? 'shade.940' : 'shade.960';
+  }
+  return color;
+};
+
+const Tab = ({ label, isActive, icon, onPress, disabled, testID }) => {
+  const initialLabelColor = getLabelColor({ isActive, disabled });
+
+  const [tabBackground, setTabBackground] = useState('transparent');
+  const [labelColor, setLabelColor] = useState(initialLabelColor);
+
+  const theme = useContext(ThemeContext);
+
+  const onPressIn = () => {
+    if (isActive) {
+      setLabelColor('primary.900');
+      setTabBackground(getColor(theme, 'primary.200'));
+      return;
+    }
+    setLabelColor('shade.970');
+    setTabBackground(getColor(theme, 'tone.940'));
+  };
+
+  const onPressOut = () => {
+    const color = getLabelColor({ isActive, disabled });
+    setLabelColor(color);
+    setTabBackground('transparent');
+  };
+
+  const isValidIcon = icons[icon];
```

We probably don't need this check. If it's not a valid icon, let it crash instead of hiding the icon

---

Yeah, wanted to have a discussion on this! Is is not good to render nothing for invalid icon rather than forcing a crash?

---

It's better to crash and let the implementor know something is wrong rather than hiding it and letting the implementor debug what went wrong